beacon_node/builder_client/src/lib.rs

```diff
@@ -135,10 +134,10 @@ impl BuilderHttpClient {
     }
 
     /// `POST /eth/v1/builder/blinded_blocks`
-    pub async fn post_builder_blinded_blocks<E: EthSpec>(
+    pub async fn post_builder_blinded_blocks<E: EthSpec, Payload: DeserializeOwned>(
```

Comment:
I was thinking to create a `trait` for `ExecutionPayload` and `ExecutionPayloadAndSidecar` to restrict the bound, but having some issues with getting this to compile

```
pub trait BuilderExecutionPayloadResponse: DeserializeOwned {}
impl<T: EthSpec> BuilderExecutionPayloadResponse for ExecutionPayloadAndBlobsSidecar<T> {}
impl<T: EthSpec> BuilderExecutionPayloadResponse for ExecutionPayload<T> {}
```